File: programers_Lv1/음양 더하기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/76501?language=javascript
function solution(absolutes, signs) {
  const answer = signs.reduce((acc, curr, idx) => {
    acc += curr ? absolutes[idx] : -absolutes[idx];
    return acc;
  }, 0);

  return answer;
}

console.log(solution([4, 7, 12], [true, false, true]));

File: programers_Lv1/음양 더하기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/76501?language=python3
def solution(absolutes, signs):
    answer = 0
    for i, v in enumerate(signs):
        answer += -absolutes[i] if not v else absolutes[i]

    return answer


print(solution([4, 7, 12], [True, False, True]))
